build.gradle.kts

```diff
@@ -23,28 +24,53 @@ plugins {
     id("java-library")
     id("spotless-java-extension")
     id("maven-publish")
+    id("signing")
     alias(libs.plugins.errorprone)
     alias(libs.plugins.buildconfig)
+    alias(libs.plugins.nexus.publish)
 }
 
 repositories { mavenCentral() }
 
 java {
-    toolchain { languageVersion = JavaLanguageVersion.of(17) }
+    toolchain { languageVersion = JavaLanguageVersion.of(17) } // Remember to update javadoc links
     withJavadocJar()
     withSourcesJar()
 }
 
-tasks.withType<Javadoc> { exclude("/com/mongodb/hibernate/internal/**") }
+tasks.withType<Javadoc> {
+    exclude("/com/mongodb/hibernate/internal/**")
+    exclude("com/mongodb/hibernate/dialect/**")
+    exclude("com/mongodb/hibernate/jdbc/**")
+
+    val standardDocletOptions = options as StandardJavadocDocletOptions
+    standardDocletOptions.apply {
+        author(true)
+        version(true)
+        encoding = "UTF-8"
+        charSet("UTF-8")
+        docEncoding("UTF-8")
+        addBooleanOption("html5", true)
+        addBooleanOption("-allow-script-in-comments", true)
+        links =
+            listOf(
+                "https://docs.oracle.com/en/java/javase/17/docs/api/",
+                "https://docs.jboss.org/hibernate/orm/6.6/javadocs/",
+                "https://mongodb.github.io/mongo-java-driver/5.3/apidocs/mongodb-driver-sync/")
```

Comment:
- I noticed that we are missing links to Jakarta Persistence, BSON, driver core.
- https://docs.jboss.org/hibernate/orm/6.6/javadocs/ is redirected to a different URI, and `javadoc` complaints about the redirect.

Adding the missing links requires a bit more meddling. I implemented what's needed in 78acc908aa1831faa3cb4ec208692dfbe21b70fa, and also created [HIBERNATE-129: Disable the modularity mismatch javadoc warning](https://jira.mongodb.org/browse/HIBERNATE-129), which we cannot do for now.

Reply:
I've merged this in, although I can't see where the links to 'Jakarta Persistence' are 🔍 - Just want to clarify its actually needed.